drivers/soundwire/stream.c

```diff
@@ -1374,12 +1374,12 @@ int sdw_stream_add_slave(struct sdw_slave *slave,
 		goto stream_error;
 	}
 
+	list_add_tail(&s_rt->m_rt_node, &m_rt->slave_rt_list);
+
```

Comment:
I prefer adding a kfree(s_rt) below line 1378 (in the initial code)

The reason is that we would add an unconfigured node to the list and then remove it. It's better to keep that list clean and avoid a useless list manipulation.

it's equivalent in the end but less is more.

Also please add a 

````
Fixes: bbe7379d8040a ("soundwire: Add support for SoundWire stream management")

````

Reply:
thanks, I will refine it.